docs/library/index.rst

```diff
@@ -181,23 +186,48 @@ The following libraries are specific to the Zephyr port.
 
   zephyr.rst
 
+.. _micropython_lib_extending:
+
 Extending built-in libraries from Python
 ----------------------------------------
 
-In most cases, the above modules are actually named ``umodule`` rather than
-``module``, but MicroPython will alias any module prefixed with a ``u`` to the
-non-``u`` version. However a file (or :term:`frozen module`) named
-``module.py`` will take precedence over this alias.
+Many built-in modules are actually named ``umodule`` rather than ``module``, but
+MicroPython will alias any module prefixed with a ``u`` to the non-``u``
+version. This means that, for example, ``import time`` will first attempt to
+resolve from the filesystem, and then failing that will fall back to the
+built-in ``utime``. On the other hand, ``import utime`` will always go directly
+to the built-in.
 
 This allows the user to provide an extended implementation of a built-in library
-(perhaps to provide additional CPython compatibility). The user-provided module
-(in ``module.py``) can still use the built-in functionality by importing
-``umodule`` directly. This is used extensively in :term:`micropython-lib`. See
-:ref:`packages` for more information.
-
-This applies to both the Python standard libraries (e.g. ``os``, ``time``, etc),
-but also the MicroPython libraries too (e.g. ``machine``, ``bluetooth``, etc).
-The main exception is the port-specific libraries (``pyb``, ``esp``, etc).
+(perhaps to provide additional CPython compatibility or missing functionality).
+The user-provided module (in ``module.py``) can still use the built-in
+functionality by importing ``umodule`` directly (e.g. typically an extension
+module ``time.py`` will do ``from utime import *``). This is used extensively
+in :term:`micropython-lib`. See :ref:`packages` for more information.
+
+This applies to the following Python standard library modules: ``array``,
+``binascii``, ``collections``, ``errno``, ``hashlib``, ``heapq``, ``io``,
+``json``, ``os``, ``platform``, ``random``, ``re``, ``select``, ``socket``,
+``ssl``, ``struct``, ``sys``, ``time``, ``zlib``, as well as the
+MicroPython-specific libraries: ``bluepy``, ``bluetooth``, ``cryptolib``,
+``ctypes``, ``machine``, ``timeq``, ``websocket``. All other built-in modules
```

Comment:
IMO `ucryptolib` and `uctypes` are different and should not be listed here, but I can see the advantage/simplicity of just listing everything that starts with u.